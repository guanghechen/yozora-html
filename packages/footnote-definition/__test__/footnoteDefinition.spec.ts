import type {
  FootnoteDefinition as IFootnoteDefinition,
  Text as IText,
} from '@yozora/ast'
import { renderChildren } from '../../../jest.setup'
import renderFootnoteDefinition from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const node: IFootnoteDefinition = {
      type: 'footnoteDefinition',
      identifier: 'bravo',
      label: 'Bravo',
      children: [
        {
          type: 'text',
          value: 'bravo and charlie.',
        } as IText,
      ],
    }

    expect(renderFootnoteDefinition(node, renderChildren)).toMatchSnapshot()
  })

  it('basic-2', function () {
    const nodes: IFootnoteDefinition[] = [
      {
        type: 'footnoteDefinition',
        identifier: 'bravo"',
        label: 'bravo<a>"<b:"></sup>',
        children: [
          {
            type: 'text',
            value: 'bravo and charlie.',
          } as IText,
        ],
      },
      {
        type: 'footnoteDefinition',
        identifier: 'bravo"',
        label: 'bravo<a>"<b:"></sup>',
        children: [],
      },
    ]

    for (const node of nodes) {
      expect(renderFootnoteDefinition(node, renderChildren)).toMatchSnapshot()
    }
  })
})
