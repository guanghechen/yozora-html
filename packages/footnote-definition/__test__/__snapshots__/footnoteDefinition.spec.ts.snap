// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `"<li id=bravo class="yozora-footnote-definition"><p class="yozora-footnote-definition__title yozora-paragraph"><a href=#reference-bravo>&uarr;</a><span>Bravo</span></p><div class="yozora-footnote-definition__content"><span class="yozora-text">bravo and charlie.</span></div></li>"`;

exports[`snapshot basic-2 1`] = `"<li id=bravo%22 class="yozora-footnote-definition"><p class="yozora-footnote-definition__title yozora-paragraph"><a href=#reference-bravo%22>&uarr;</a><span>bravo"</span></p><div class="yozora-footnote-definition__content"><span class="yozora-text">bravo and charlie.</span></div></li>"`;

exports[`snapshot basic-2 2`] = `"<li id=bravo%22 class="yozora-footnote-definition"><p class="yozora-footnote-definition__title yozora-paragraph"><a href=#reference-bravo%22>&uarr;</a><span>bravo"</span></p><div class="yozora-footnote-definition__content"></div></li>"`;
